### PR TITLE
feat: Make as Variant optional for #[messages(...)] attribtue

### PR DESCRIPTION
**Pull request by kulikthebird**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -385,7 +390,9 @@ matching the trait name, but written "snake_case" instead of CamelCase. Here I h
 `group` module for the `Group` trait, but the `CrossStaking` trait should be placed
 in its own `cross_staking` module (note the underscore). This is a requirement right
 now - Sylvia generates all the messages and boilerplate in this module and will try
-to access them through this module.
+to access them through this module. If the interface's name is a camel-case
+version of the last module path's segment, the `as InterfaceName` can be omitted.
```
Comment (jawoznia) [inline README.md:394]:
```suggestion
to access them through this module. If the interface's name is a camel-case
version of the last module path's segment, the `as InterfaceName` can be omitted.
F.e. `#[messages(cw1 as Cw1)]` can be reduced to `#[messages(cw1)]`
```

File: sylvia-derive/src/parser.rs [hunk 2]
```diff
@@ -349,18 +350,45 @@ impl Parse for ContractMessageAttr {
...
+                == module
+                    .segments
+                    .last()
+                    .map(|name| name.ident.to_string().to_case(Case::UpperCamel))
+            {
+                emit_warning!(
+                    variant.span(),
+                    "Interface name `as {}` can be omitted, since it's corresponds to the path's last segment.",
+                    variant
+                )
```
Comment (jawoznia) [inline sylvia-derive/src/parser.rs:365]:
```suggestion
                let msg = format! { "Redundant `as {}`." };
                emit_warning!(
                    variant.span(), msg,
                    note = "Interface name is a camel case version of the path and can be auto deduced.",
                    variant
                )
```

**Review by jawoznia — APPROVED** [review]
Consider better error messages. Otherwise LGTM.

File: sylvia-derive/src/parser.rs [hunk 3]
```diff
@@ -349,18 +350,45 @@ impl Parse for ContractMessageAttr {
...
+                        let err =
+                            "#`[message(..)]` attribute without `as TraitName` needs to provide a path";
+                        emit_error!(module.span(), err);
+                        syn::Error::new(module.span(), err)
+                    })?
+                    .ident
+                    .to_string()
+                    .to_case(Case::UpperCamel),
+                    module.span()
+            )
```
[the remark this answers is not in the document]
Comment (kulikthebird) [reply sylvia-derive/src/parser.rs]: Done

**Pull request merged**